pyiceberg/expressions/__init__.py

```diff
@@ -559,8 +561,10 @@ def as_bound(self) -> Type[BoundNotNaN[L]]:
         return BoundNotNaN[L]
 
 
-class SetPredicate(UnboundPredicate[L], ABC):
-    literals: Set[Literal[L]]
+class SetPredicate(UnboundPredicate[L], IcebergBaseModel, ABC):
+    type: str = Field(default="in", alias="type")
+    term: str
```

Comment:
This one is actually defined in the parent class:

https://github.com/apache/iceberg-python/blob/6913b11ae1abbe0cfb05bfb0232fe66f6de437be/pyiceberg/expressions/__init__.py#L417

```suggestion
```